pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -364,13 +407,9 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env model.En
 
 	var tcpListeners, httpListeners []*xdsapi.Listener
 	var currentListener *xdsapi.Listener
-	listenerTypeMap := make(map[string]model.Protocol)
-	listenerMap := make(map[string]*xdsapi.Listener)
+	listenerMap := make(map[string]listenerEntry)
```

Comment:
can we use *listenerEntry ? No need to make a copy of the objec.t

Reply:
I was trying to avoid a bunch of small heap allocations.  The struct is quite small.